src/dbnode/storage/database.go

```diff
@@ -879,40 +885,122 @@ func (d *db) ReadEncoded(
 			xopentracing.Time("end", end),
 		)
 	}
+
 	defer sp.Finish()
+	return n.ReadEncoded(ctx, id, start, end)
+}
 
+func (d *db) WideQuery(
+	ctx context.Context,
+	namespace ident.ID,
+	query index.Query,
+	start time.Time,
+) error { // FIXME: change when exact type known.
```

Comment:
// iterator for these: (ident.BytesID, []byte, []byte, int, error)
// r.streamingID, r.streamingTags, r.streamingData, shardID, error